src/neuroglancer/ui/tool.ts

```diff
@@ -258,19 +258,29 @@ export class ToolBinder extends RefCounted {
     }
     this.debounceDeactivate.cancel();
     if (tool === this.activeTool?.tool) {
+      if (tool.toggle) {
+        this.deactivate();
+      }
+      return;
+    }
+    else if (this.activeTool) {
+      StatusMessage.showTemporaryMessage(
+        `Can't activate tool ${tool.description} while tool ${this.activeTool.tool.description} is active`);
```

Comment:
Perhaps it would be better to just deactivate the existing tool in this case?

Another question is whether there should now be a way to encode in the JSON state which tool is activated.

Reply:
There has been some discussion of this. An example scenario is a multicut in progress, where the user does a significant amount of work with the tool active, and may not want it to be easily deactivated by accident. That could be mitigated by not clearing the multicut state when it's deactivated, but that may be confusing. We're still discussing the ideal behavior from a user's perspective for this.

Reply:
I think there definitely should be a way to deactivate without clearing the state, because the user may wish to use another tool to adjust display settings, etc.

Possibly when you activate a non-toggle tool while a toggle tool is activated, the toggle tool should be temporarily deactivated, but then re-activated once there are no non-toggle tools activated.  I would suppose that activating another toggle tool should permanently deactivate the previous toggle tool.

As far as multicut, my understanding is that it was specifically desired to be able to share the multicut points by sharing the neuroglancer state, and that was the reason for encoding the multicut points in the neuroglancer state in the first place.  That means at the very least either the toggled-on tool needs to be encoded in the JSON state, or the multicut points need to be kept even when the tool is deactivated.

Reply:
The team agreed the behavior you suggest is good, so I pushed a change to make it work that way. Now, if a toggle tool is active and another toggle tool is activated, the first one gets disabled and the new one becomes active. If a toggle tool is active and a non-toggle tool is activated, the first one gets disabled and the new one becomes active, with the first one becoming queued to reactivate as soon as the new one is deactivated. So from a user's perspective, you can only have one toggle tool at a time, but activate any non-toggle tool "on top of" that active tool. (Also, multicut points will be preserved when the tool is disabled, but the graphene changes aren't included in this PR.)